staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -685,9 +685,15 @@ func (s *store) watch(ctx context.Context, key string, rv string, pred storage.S
 
 func (s *store) getState(getResp *clientv3.GetResponse, key string, v reflect.Value, ignoreNotFound bool) (*objState, error) {
 	state := &objState{
-		obj:  reflect.New(v.Type()).Interface().(runtime.Object),
 		meta: &storage.ResponseMeta{},
 	}
+
+	if u, ok := v.Addr().Interface().(runtime.Unstructured); ok {
```

Comment:
check using `CanAddr()`? test seems to be panic'ing

```
--- FAIL: TestCreateWithKeyExist (0.10s)
panic: reflect.Value.Addr of unaddressable value [recovered]
	panic: reflect.Value.Addr of unaddressable value
```

Reply:
all real places that call this use `conversion.EnforcePtr(...)` which ensures addressability

Reply:
switched the test to use that as well